api/src/main/java/org/pmiops/workbench/reporting/ReportingVerificationServiceImpl.java

```diff
@@ -37,19 +39,71 @@
 
   private final BigQueryService bigQueryService;
   private final Provider<WorkbenchConfig> workbenchConfigProvider;
-  private Provider<Stopwatch> stopwatchProvider;
+  private final Provider<Stopwatch> stopwatchProvider;
+  private final ReportingQueryService reportingQueryService;
 
   public ReportingVerificationServiceImpl(
       BigQueryService bigQueryService,
       Provider<WorkbenchConfig> workbenchConfigProvider,
-      Provider<Stopwatch> stopwatchProvider) {
+      Provider<Stopwatch> stopwatchProvider,
+      ReportingQueryService reportingQueryService) {
     this.bigQueryService = bigQueryService;
     this.workbenchConfigProvider = workbenchConfigProvider;
     this.stopwatchProvider = stopwatchProvider;
+    this.reportingQueryService = reportingQueryService;
   }
 
   @Override
-  public ReportingUploadDetails getUploadDetails(ReportingSnapshot snapshot) {
+  public boolean verifyAndLog(ReportingSnapshot reportingSnapshot) {
+    // check each table. Note that for streaming inputs, not all rows may be immediately available.
+    final ReportingUploadDetails uploadDetails = getUploadDetails(reportingSnapshot);
+    final StringBuilder sb =
+        new StringBuilder(
+            String.format("Verifying Snapshot %d:\n", reportingSnapshot.getCaptureTimestamp()));
+    sb.append("Table\tUploaded\tExpected\tDifference(%)\n");
+    Level detailsLogLevel = Level.INFO;
+    boolean verified = true;
+    for (final ReportingUploadResult result : uploadDetails.getUploads()) {
+      if (!verifyCount(
+          result.getTableName(), result.getSourceRowCount(), result.getDestinationRowCount(), sb)) {
+        detailsLogLevel = Level.WARNING;
+        verified = false;
+      }
+    }
+    logger.log(detailsLogLevel, sb.toString());
+    return verified;
+  }
+
+  @Override
+  public boolean verifyBatchesAndLog(List<String> batchTables, long captureSnapshotTime) {
+    final StringBuilder sb =
+        new StringBuilder(String.format("Verifying batches %d:\n", captureSnapshotTime));
+    Level detailsLogLevel = Level.INFO;
+    boolean verified = true;
+
+    for (String tableName : batchTables) {
+      sb.append("Table\tUploaded\tExpected\tDifference(%)\n");
+      switch (tableName) {
+        case WorkspaceColumnValueExtractor.TABLE_NAME:
```

Comment:
Since this only executes for one table, let's change this to `if (batchTables.contains(WorkspaceColumnValueExtractor.TABLE_NAME))`